File: spark-scala/src/main/scala/com/scala/learn/spark/datareader/base/DataReader.scala
package com.scala.learn.spark.datareader.base

import com.scala.learn.spark.base.ScalaSparkSessionProvider
import org.apache.spark.sql.{DataFrameReader, Dataset, Row};

abstract class DataReader extends ScalaSparkSessionProvider {

  val dataFrameReader: DataFrameReader = spark.read.option("header", "true").option("inferSchema", "true")

  def getData(path: String): Dataset[Row]
}

File: spark-java/src/main/java/com/learn/spark/datareader/base/DataReader.java
package com.learn.spark.datareader.base;

import com.learn.spark.base.JavaSparkSessionProvider;
import org.apache.spark.sql.DataFrameReader;
import org.apache.spark.sql.Dataset;

public interface DataReader<T>  {

    default DataFrameReader getDataReader(){
        return JavaSparkSessionProvider.spark.read();
    }

    Dataset<T> getData();

}
